mediator/tenants/models.py

```diff
@@ -0,0 +1,38 @@
+from cryptography.fernet import Fernet
+from django.conf import settings
+from django.contrib.auth.models import AbstractUser
+from django.db import models
+
+
+class Tenant(models.Model):
+    short_name = models.SlugField(max_length=200, unique=True)
+    name = models.CharField(max_length=200)
+
+
+class Upstream(models.Model):
+    """
+    An API upstream of this mediator. e.g. DHIS2 or OpenMRS
+    """
+    tenant = models.ForeignKey(Tenant, on_delete=models.CASCADE)
+    short_name = models.SlugField(max_length=200, db_index=True)
+    base_url = models.URLField()
+    verify_cert = models.BooleanField(default=True)
+    username = models.CharField(max_length=200)
+    # NOTE: If settings.SECRET_KEY is changed, passwords cannot be decrypted
+    _password = models.CharField(max_length=200)
```

Comment:
this is neat. did you model it after `auth.User`?

Reply:
Yes, and [`motech.Repeater`](https://github.com/dimagi/commcare-hq/blob/efa37afb07628284bb2234797dd648b245dfe74b/corehq/motech/repeaters/models.py#L278). 

(`auth.User` is fancier though, cos it supports multiple [types of hashes](https://docs.djangoproject.com/en/2.1/ref/settings/#std:setting-PASSWORD_HASHERS), and uses [`set_password()`](https://github.com/django/django/blob/master/django/contrib/auth/base_user.py#L97) and `check_password()` methods instead of using a property. (It also [automatically upgrades](https://docs.djangoproject.com/en/2.1/topics/auth/passwords/#password-upgrading) a user's password hash on login; pretty cool.))